Fix download log test

nivi-apple opened the pull request:
Fix the download log test that expects a failure by waiting for the status report before continuing to the next test

Fixes: #32636

harimau-qirex commented on src/app/clusters/diagnostic-logs-server/BDXDiagnosticLogsProvider.cpp:
```diff
@@ -230,14 +230,14 @@ void BDXDiagnosticLogsProvider::OnInternalError()
     ChipLogError(BDX, "Timeout");
     VerifyOrDo(mIsAcceptReceived, SendCommandResponse(StatusEnum::kDenied));
-    Reset();
+    mTransfer.AbortTransfer(GetBdxStatusCodeFromChipError(CHIP_ERROR_TIMEOUT));
```
How does the transfer get reset after this point (and similarly in OnInternalError)? My understanding is AbortTransfer will trigger a status report to be sent and then the exchange will close since no response is expected for the status report. I don't think this class calls Reset when handling those (through OnMsgToSend or OnExchangeClosing), so would the rest of the internal state still be set?

nivi-apple replied:
Yeah I guess you are right. I looked at BDX a long time ago. From what I remember for OTA we abort transfer and then waited sometime for the status report to go through and called Reset from what I recall. We could something similar. That being said I am not sure if the test is failing due to a genuine issue or merely not cleaning up after finishing the tests and overlaps the next test. I am looking at that now. will update the PR accordingly

nivi-apple committed:
Add a check for Diagnostic logs success message after test - "Read Network Diagnostic log intent" to make sure the test completes.

- Add a check for Diagnostic logs status report message after test - "Read Network Diagnostic log intent with a very short timeout and a
  very long log" to make sure the test is complete.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/project-chip/connectedhomeip?pullRequest=33788) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/project-chip/connectedhomeip?pullRequest=33788) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/project-chip/connectedhomeip?pullRequest=33788) it.</sub>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/project-chip/connectedhomeip?pullRequest=33788) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/project-chip/connectedhomeip?pullRequest=33788) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/project-chip/connectedhomeip?pullRequest=33788) it.</sub>
